iym/admin.py

```diff
@@ -0,0 +1,348 @@
+        headers=authorize_upload_headers,
+    )
+
+    r.raise_for_status()
+    return r.json()
+
+
+def upload(path, authorisation):
+    # Unpack values out of lists
+    upload_query = {k: v[0] for k, v in authorisation["upload_query"].items()}
+    upload_url = f"{authorisation['upload_url']}?{urlencode(upload_query)}"
+    upload_headers = {
+        "content-type": "application/octet-stream",
+        "Content-MD5": authorisation["md5"]
+    }
+    with open(path, 'rb') as file:
+        r = requests.put(upload_url, data=file, headers=upload_headers)
+    r.raise_for_status()
+
+
+def unzip_uploaded_file(obj_to_update):
+    relative_path = 'iym/temp_files/'
+    zip_file = obj_to_update.file
+
+    with ZipFile(zip_file, 'r') as zip:
+        file_name = zip.namelist()[0]
+        zip.extractall(path=relative_path)
+
+    original_csv_path = os.path.join(settings.BASE_DIR, relative_path, file_name)
+
+    return original_csv_path
+
+
+def create_composite_key_using_csv_headers(original_csv_path):
+    # Get all the headers to come up with the composite key
+    with open(original_csv_path) as original_csv_file:
+        reader = csv.DictReader(original_csv_file)
+        first_row = next(reader)
+        fields = first_row.keys()
+        composite_key = list(set(fields) - set(MEASURES))
+
+    return composite_key
+
+
+def tidy_csv_table(original_csv_path, composite_key):
+    table1 = etl.fromcsv(original_csv_path)
+    table2 = etl.convert(table1, MEASURES, lambda v: v.replace(",", "."))
+    table3 = etl.convert(table2, "Financial_Year", lambda v: RE_END_YEAR.sub("", v))
+    table4 = etl.convert(table3, MEASURES, Decimal)
+
+    # Roll up rows with the same composite key into one, summing values together
+    # Prefixing each new measure header with "sum" because petl seems to need
+    # different headers for aggregation output
+    aggregation = {f"sum{measure}": (measure, sum) for measure in MEASURES}
+    table5 = etl.aggregate(table4, composite_key, aggregation)
+
+    # Strip sum prefix from aggregation results
+    measure_rename = {key: key[3:] for key in aggregation}
+    table6 = etl.rename(table5, measure_rename)
+
+    return table6
+
+
+def create_data_package(csv_filename, csv_table, userid, data_package_name, data_package_title, obj_to_update):
+    data_package_template_path = "iym/data_package/data_package.json"
+    base_token = settings.OPEN_SPENDING_BASE_TOKEN
+
+    with tempfile.NamedTemporaryFile(mode="w", delete=False) as csv_file:
+        csv_path = csv_file.name
+        etl.tocsv(csv_table, csv_path)
+        update_import_report(obj_to_update, "Getting authorisation for datastore")
+
+        authorize_query = {
+            "jwt": base_token,
+            "service": "os.datastore",
+            "userid": userid,
+        }
+        authorize_url = f"https://openspending-dedicated.vulekamali.gov.za/user/authorize?{urlencode(authorize_query)}"
+        print('============ aaa ============')
```

Comment:
remove print debugging

Reply:
this was a test commit - print is removed now